fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/DeleteResourceServiceImpl.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.kernel.impl.services;
+
+import org.fcrepo.kernel.api.ContainmentIndex;
+import org.fcrepo.kernel.api.Transaction;
+import org.fcrepo.kernel.api.exception.PathNotFoundException;
+import org.fcrepo.kernel.api.exception.PathNotFoundRuntimeException;
+import org.fcrepo.kernel.api.exception.RepositoryRuntimeException;
+import org.fcrepo.kernel.api.models.Binary;
+import org.fcrepo.kernel.api.models.Container;
+import org.fcrepo.kernel.api.models.FedoraResource;
+import org.fcrepo.kernel.api.models.NonRdfSourceDescription;
+import org.fcrepo.kernel.api.models.ResourceFactory;
+import org.fcrepo.kernel.api.operations.DeleteResourceOperationFactory;
+import org.fcrepo.kernel.api.operations.ResourceOperation;
+import org.fcrepo.kernel.api.services.DeleteResourceService;
+import org.fcrepo.persistence.api.PersistentStorageSession;
+import org.fcrepo.persistence.api.PersistentStorageSessionManager;
+import org.fcrepo.persistence.api.exceptions.PersistentStorageException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.util.stream.Stream;
+
+import static java.lang.String.format;
+
+/**
+ * This class mediates delete operations between the kernel and persistent storage layers
+ *
+ * @author dbernstein
+ */
+public class DeleteResourceServiceImpl implements DeleteResourceService {
+
+    private final static Logger log = LoggerFactory.getLogger(DeleteResourceService.class);
+
+    @Inject
+    private ContainmentIndex containmentIndex;
+
+    @Inject
+    private ResourceFactory resourceFactory;
+
+    @Inject
+    private DeleteResourceOperationFactory deleteResourceFactory;
+
+    @Inject
+    private PersistentStorageSessionManager psManager;
+
+    @Override
+    public void perform(final Transaction tx, final FedoraResource fedoraResource) {
+        final String fedoraResourceId = fedoraResource.getId();
+
+        if( fedoraResource instanceof NonRdfSourceDescription){
```

Comment:
the space missed :( should be `if (` and there should be a space after the closing `)`. Are you using a code formatting rules in your IDE? I know they exist for eclipse, not sure how intellij differs

Reply:
done.